src/frontends/tensorflow/src/helper_ops/gru_block_cell.hpp

```diff
@@ -0,0 +1,115 @@
+        : InternalOperation(decoder, OutputVector{x, h_prev, w_ru, w_c, b_ru, b_c}, 4),
+          m_hidden_size(ov::Dimension::dynamic()) {
+        validate_and_infer_types();
+    }
+
+    void validate_and_infer_types() override {
+        // GRUBlockCell computes the GRU cell forward propagation for 1 time step
+        // Inputs:
+        // 0) x: Input to the GRU cell
+        // 1) h_prev: State input from the previous GRU cell
+        // 2) w_ru: Weight matrix for the reset and update gate
+        // 3) w_c: Weight matrix for the cell connection gate
+        // 4) b_ru: Bias vector for the reset and update gate
+        // 5) b_c: Bias vector for the cell connection gate
+        //
+        // Outputs:
+        // 0) r: Output of the reset gate
+        // 1) u: Output of the update gate
+        // 2) c: Output of the cell connection gate
+        // 3) h: Current state of the GRU cell
+
+        // try to deduce static hidden_size
+        // 1. use h_prev shape
+        auto h_prev_shape = get_input_partial_shape(1);
+        auto h_prev_rank = h_prev_shape.rank();
+        if (h_prev_rank.is_static()) {
+            FRONT_END_OP_CONVERSION_CHECK(h_prev_rank.get_length() == 2,
+                                          "Internal error in OpenVINO TensorFlow Frontend: initial hidden state for "
+                                          "GRUBlockCell must be of rank equal to 2.");
+            m_hidden_size = h_prev_shape[1].is_static() ? h_prev_shape[1].get_length() : m_hidden_size;
+        }
+        // 2. use w_ru shape
+        auto w_ru_shape = get_input_partial_shape(2);
+        auto w_ru_rank = w_ru_shape.rank();
+        if (w_ru_rank.is_static()) {
+            FRONT_END_OP_CONVERSION_CHECK(
+                w_ru_rank.get_length() == 2,
+                "Internal error in OpenVINO TensorFlow Frontend: weights for GRUBlockCell must be of rank equal to 2.");
+            m_hidden_size = w_ru_shape[1].is_static() ? w_ru_shape[1].get_length() / 2 : m_hidden_size;
+        }
+        // 3. use w_c shape
+        auto w_c_shape = get_input_partial_shape(3);
+        auto w_c_rank = w_c_shape.rank();
+        if (w_c_rank.is_static()) {
+            FRONT_END_OP_CONVERSION_CHECK(
+                w_c_rank.get_length() == 2,
+                "Internal error in OpenVINO TensorFlow Frontend: weights for GRUBlockCell must be of rank equal to 2.");
+            m_hidden_size = w_c_shape[1].is_static() ? w_c_shape[1].get_length() : m_hidden_size;
+        }
+        // 3. use b_ru shape
+        auto b_ru_shape = get_input_partial_shape(4);
+        auto b_ru_rank = b_ru_shape.rank();
+        if (b_ru_rank.is_static()) {
+            FRONT_END_OP_CONVERSION_CHECK(
+                b_ru_rank.get_length() == 1,
+                "Internal error in OpenVINO TensorFlow Frontend: bias for GRUBlockCell must be of rank equal to 1.");
+            m_hidden_size = b_ru_shape[0].is_static() ? b_ru_shape[0].get_length() / 2 : m_hidden_size;
```

Comment:
what if b_ru_shape[0].get_length() / 2 is not equal the current value of m_hidden_size?
Maybe we need to add an additional check in every if

Reply:
Discussed offline